src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -8410,6 +8429,487 @@ namespace WaterThermalTanks {
+                                AvgFactorMixing = ((exp(A[k] * dt) - 1.0) / A[k] / dt - 1.0) / A[k] / NodeCapacitance;
+                            }
+                            const Real64 Q_AdiabaticMixing = (Tmixed - Tfinal[k]) / FinalFactorMixing;
+                            Tfinal[k] = Tmixed;
+                            Tavg[k] += Q_AdiabaticMixing * AvgFactorMixing;
+                        }
+
+                        // Since we mixed, get out of here and start from the top to check again for mixing.
+                        break;
+                    }
+                }
+            } while ( HasInversion );
+
+            // Venting
+            if (Tfinal[0] > Tank.TankTempLimit) {
+                for (int i = 0; i < nTankNodes; ++i) {
+                    if (Tfinal[i] > Tank.TankTempLimit) {
+                        Event += Tank.Node[i].Mass * Cp * (Tank.TankTempLimit - Tfinal[i]);
+                        Tfinal[i] = Tank.TankTempLimit;
+                    }
+                }
+            }
+
+            // Increment to next internal time step
+            TimeRemaining -= dt;
+            Real64 Qloss = 0.0;
+            for (int i = 0; i < nTankNodes; ++i) {
+                auto &node = Tank.Node[i];
+                node.Temp = Tfinal[i];
+                node.TempSum += Tavg[i] * dt;
+
+                // Bookkeeping for reporting variables
+                if (Tank.HeaterOn1 || Tank.HeaterOn2) {
+                    Qloss += node.OnCycLossCoeff * (Tank.AmbientTemp - Tavg[i]);
+                } else {
+                    Qloss += node.OffCycLossCoeff * (Tank.AmbientTemp - Tavg[i]);
+                }
+            }
+            // More bookkeeping for reporting variables
+            Eloss += Qloss * dt;
+            const Real64 Quse = Tank.UseEffectiveness * Tank.UseMassFlowRate * Cp * (Tank.UseInletTemp - Tavg[Tank.UseOutletStratNode - 1]);
+            Euse += Quse * dt;
+            const Real64 Qsource = [&]{
+                if (Tank.HeatPumpNum > 0) {
+                    if (HPWHCondenserConfig == TypeOf_HeatPumpWtrHeaterPumped) {
+                        return Qheatpump;
+                    } else {
+                        assert(HPWHCondenserConfig == TypeOf_HeatPumpWtrHeaterWrapped);
+                        return 0.0;
+                    }
+                } else {
+                    return Tank.SourceEffectiveness * Tank.SourceMassFlowRate * Cp * (Tank.SourceInletTemp - Tavg[Tank.SourceOutletStratNode - 1]);
+                }
+            }();
+            Esource += Qsource * dt;
+            if (Tank.HeaterOn1) Runtime1 += dt;
+            if (Tank.HeaterOn2) Runtime2 += dt;
+            if (Tank.HeaterOn1 || Tank.HeaterOn2) Runtime += dt;
+            Eheater1 += Qheater1 * dt;
+            Eheater2 += Qheater2 * dt;
+            Eunmet += max(Tank.UseEffectiveness * Tank.UseMassFlowRate * (UseOutletExpectedTemp - Tavg[Tank.UseOutletStratNode - 1]), 0.0) * dt;
```

Comment:
@mjwitte This is being calculated differently that before. That's intentional. This is measuring amount of additional heat needed in the outlet stream to meet setpoint. 

The way it was being calculated before didn't make sense. (For example, it was accounting for the heaters in every node, not just the ones with the heaters). There wasn't anything in the documentation about this output and what its physical definition was. This seemed more defensible.

Reply:
Hmm, the load on a WH is not only equal to the local load at the heater node but the load in the rest of the tank (thought, where does mixing occur?). So if a node does not have an element, and the dT is < 0 then isn't there a load on the node with an element? Just because a node doesn't have an element doesn't mean there is no load on nodes that do have an element. It only means Q=0 at that node. After saying this, if there are 2 elements in a tank with 2+ nodes, how is the tank load split between elements (i.e., if dT is used in line 8826 then which element does that Q apply to)?

Reply:
The previous calcs (before these changes) take into account the total load. Take these calcs for instance. Don't these look at a total tank load based on use/source/loss loads? That's not a load only at node that have a heater. So maybe integrating nodes without a heater may be exactly what the model should do.

    Qheat = Qoncycheat = tank_node.OnCycParaLoad * Tank.OnCycParaFracToTank;
    Qusl = Quse + Qsource + Qloss;
    Qneeded = max(-Qusl - Qoncycheat, 0.0);
    Qunmet = max(Qneeded - Qheaters, 0.0);

Reply:
@rraustad The snippet you shared above illustrates what I'm talking about. It is being calculated inside a loop of the nodes. `Qusl` has the use, source, and loss heat rates _for the node being considered_, so the `Quse` and `Qsource` are only nonzero for the use and source inlet nodes respectively. So far so good. Same for the on or off cycle heat. However, `Qheaters` is sum of the heat output of both of the heaters during the timestep. Those heaters are assigned to act on specific nodes, but in this case they're being accounted for _on every node_. I can't figure out what that's supposed to physically represent. 

Is there a definition of what this unmet load rate is supposed to mean? If so, I can write code to it. Given the lack of information or definition in the IORef, I defined it in the way that made sense to me after discussing it with our guy who does water heaters. Which is to say that the unmet load of a water heater is the difference between the energy coming out of the water heater during a draw and the energy we expect to come out of the water heater during a draw given the setpoints. If the goal is to match what came out of the old model, I can do that, but I'm just not convinced that it has any meaning whatsoever.